src/ui/viewmodels/MessageBoxViewModel.hh

```diff
@@ -0,0 +1,167 @@
+#ifndef RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#define RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#pragma once
+
+#include "ui/WindowViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+namespace viewmodels {
+
+enum class MessageBoxButtons
+{
+    OK,
+    OKCancel,
+    YesNo,
+    YesNoCancel,
+    RetryCancel,
+};
+
+enum class MessageBoxIcon
+{
+    None,
+    Info,
+    Warning,
+    Error,
+};
+
+class MessageBoxViewModel : public WindowViewModelBase
+{
+public:
+    MessageBoxViewModel() noexcept {}
+
+    MessageBoxViewModel(const std::wstring& sMessage) noexcept
+    {
+        SetMessage(sMessage);
+    }
+
+#pragma push_macro("GetMessage") // windows stole my method name
+#undef GetMessage
+    /// <summary>
+    /// The <see cref="ModelProperty" /> for the message.
+    /// </summary>
+    static StringModelProperty MessageProperty;
```

Comment:
It's not a big deal but I just want to know the reasoning why these properties are static instead of data members for future reference. I personally never used static properties and preferred auto properties, I can probably make something like later. static property just seems interesting.

Reply:
Full .NET WPF must be different than .NET Core's version. Thanks for the explanation.